src/main/scala/analysis/solvers/IDESolver.scala

```diff
@@ -0,0 +1,279 @@
+          val e3 = x(funexit, d1, d2) // summary edge from d1 to d2 at the callee function
+          val e123 = e3.composeWith(e12)
+          edgesExitToAfterCall(funexit, aftercall)(d2).foreach {
+            case (d4, e4) => // d4 is now an item at the aftercall node, and e4 is the edge from the function exit to the aftercall node
+              val e = e4.composeWith(e123) // e is now the composed edge from e3 at the caller entry to d4 at the aftercall node
+              propagate(e, (aftercall, d3, d4))
+          }
+      }
+    }
+
+
+    def process(n: (CFGPosition, DL, DL)): Unit = {
+      n match {
+        case (position, d1, d2) =>
+          val e1 = x(n)
+          if isCall(position) then // at a call node
+            val call: C = position.asInstanceOf[C]
+            val entry: E = getCallee(call)
+            val ret: R = callToReturn(call)
+
+            edgesCallToEntry(call, entry)(d2).foreach {
+              case (d3, e2) =>
+                // propagate to function entry
+                propagate(IdEdge(), (entry, d3, d3))
+                // cache the composed edge from the entry of the caller to the entry of the callee
+                storeCallJump(entry, d3, call, e2.composeWith(e1), d1)
+                // propagate existing return flow to the after-call node
+                exitJumpCache.getOrElseUpdate((entry, d3), mutable.Set[DL]()).foreach { d4 =>
+                  returnflow(d3, d4, entryToExit(entry), ret)
+                }
+            }
+
+            edgesCallToAfterCall(call, ret)(d2).foreach {
+              case (d3, e2) =>
+                propagate(e2.composeWith(e1), (ret, d1, d3))
+            }
+          else if isExit(position) then // at an exit node
+            val exit: EE = position.asInstanceOf[EE]
+            getAfterCalls(exit).foreach(afterCall => returnflow(d1, d2, exit, afterCall))
+            storeExitJump(exitToEntry(exit), d1, d2)
+          else
+            edgesOther(position)(d2).foreach {
+              case (d3, e2) =>
+                val e3 = e2.composeWith(e1)
+                outdep(position).foreach { m =>
+                  propagate(e3, (m, d1, d3))
+                }
+            }
+      }
+    }
+
+    def summaries(): mutable.Map[Procedure, mutable.Map[DL, mutable.Map[DL, EdgeFunction[T]]]] = {
+      val res = mutable.Map[Procedure, mutable.Map[DL, mutable.Map[DL, EdgeFunction[T]]]]()
+      x.foreach {
+        case ((n, d1, d2), e) =>
+          if isExit(n) then
+            val exit: EE = n.asInstanceOf[EE]
+            val proc = IRWalk.procedure(exit)
+            val m1 = res.getOrElseUpdate(proc, mutable.Map[DL, mutable.Map[DL, EdgeFunction[T]]]().withDefaultValue(mutable.Map[DL, EdgeFunction[T]]()))
+            val m2 = m1.getOrElseUpdate(d1, mutable.Map[DL, EdgeFunction[T]]())
+            m2 += d2 -> e
+
+      }
+      Logger.info(s"Function summaries:\n${
```

Comment:
This is a bit too much information for the `info` log level, can we make it debug instead?